pkg/cephmgr/mon/config.go

```diff
@@ -148,6 +148,15 @@ func GenerateConfigFile(context *clusterd.Context, cluster *ClusterInfo, pathRoo
 		return "", fmt.Errorf("failed to add initial monitor config sections, %+v", err)
 	}
 
+	// if there's a config file override path given, process the given config file
+	if context.ConfigFileOverride != "" {
+		err := configFile.Append(context.ConfigFileOverride)
```

Comment:
Yes, it handles the merging of all sections and values.  Better than the implementation I started on myself before finding that Append() func :)